_overviews/scala3-book/collections-methods.md

```diff
@@ -0,0 +1,310 @@
+```
+
+
+
+## Sample lists
+
+The following examples use these lists:
+
+```scala
+val oneToTen = (1 to 10).toList
+val names = List("adam", "brandy", "chris", "david")
+```
+
+
+
+## `map`
+
+The `map` method steps through each element in the existing list, applying the algorithm you supply to each element, one at a time; it then returns a new list with all of the modified elements.
+
+Here’s an example of the `map` method being applied to the `oneToTen` list:
+
+```scala
+scala> val doubles = oneToTen.map(_ * 2)
+doubles: List[Int] = List(2, 4, 6, 8, 10, 12, 14, 16, 18, 20)
+```
+
+You can also write anonymous functions using a long form, like this:
+
+```scala
+scala> val doubles = oneToTen.map(i => i * 2)
+doubles: List[Int] = List(2, 4, 6, 8, 10, 12, 14, 16, 18, 20)
+```
+
+However, in this lesson we’ll always use the first, shorter form.
+
+Here are a few more examples of the `map` method being applied to the `oneToTen` and `names` lists:
+
+```scala
+scala> val capNames = names.map(_.capitalize)
+capNames: List[String] = List(Adam, Brandy, Chris, David)
+
+scala> val nameLengthsMap = names.map(s => (s, s.length)).toMap
+nameLengthsMap: Map[String, Int] = Map(adam -> 4, brandy -> 6, chris -> 5, david -> 5)
+
+scala> val isLessThanFive = oneToTen.map(_ < 5)
+isLessThanFive: List[Boolean] = List(true, true, true, true, false, false, false, false, false, false)
+```
+
+As shown in the last two examples, it’s perfectly legal (and common) to use `map` to return a collection that has a different type than the original type.
+
+
+
+## `filter`
+
+The `filter` method creates a new, filtered list from the given list. Here are a few examples:
+
+```scala
+scala> val lessThanFive = oneToTen.filter(_ < 5)
+lessThanFive: List[Int] = List(1, 2, 3, 4)
+
+scala> val evens = oneToTen.filter(_ % 2 == 0)
+evens: List[Int] = List(2, 4, 6, 8, 10)
+
+scala> val shortNames = names.filter(_.length <= 4)
+shortNames: List[String] = List(adam)
+```
+
+
+
+## `foreach`
+
+The `foreach` method is used to loop over all elements in a collection. Note that `foreach` is used for side-effects, such as printing information. Here’s an example with the `names` list:
+
+```scala
+scala> names.foreach(println)
+adam
+brandy
+chris
+david
+```
+
+A great thing about the functional methods on the collections classes is that you can chain them together to solve problems. For example, this is one way to print the first three elements from `oneToTen`:
+
+```scala
+oneToTen.filter(_ < 4).foreach(println)
+```
+
+The REPL shows the result:
+
+```scala
+scala> oneToTen.filter(_ < 4).foreach(println)
+1
+2
+3
+```
+
+
+
+## `head`
+
+The `head` method comes from Lisp and other earlier functional programming languages. It’s used to print the first element (the head element) of a list:
+
+```scala
+oneToTen.head   // Int = 1
+names.head      // adam
+```
+
+Because a `String` is a sequence of characters, you can also treat it like a list. This is how `head` works on these strings:
```

Comment:
```suggestion
Because a `String` can be see as a sequence of characters, you can also treat it like a list. This is how `head` works on these strings:
```